### PR TITLE
remove rebuild index guard when job finished

### PR DESCRIPTION
**Pull request by SuperYoko**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 

https://github.com/vesoft-inc/nebula-ent/issues/1410

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by critical27 — APPROVED** [review]
Can't believe we have so many problem in rebuild index ......... Good job

**Comment by SuperYoko** [comment]
> Can't believe we have so many problem in rebuild index ......... Good job

don't merge now, wait tests result later.

**Comment by SuperYoko** [comment]
> > Can't believe we have so many problem in rebuild index ......... Good job
> 
> don't merge now, wait tests result later.

Tested locally , no job will be blocked if prev job failed now.

**Comment by SuperYoko** [comment]
As you can see, failed job (caused by stop) will not affect new job, and recover is okay now.

image

**Review by liwenhui-soul — APPROVED** [review]

**Comment by pengweisong** [comment]
mark for review

File: src/storage/admin/RebuildIndexTask.cpp [hunk 1]
```diff
@@ -17,6 +17,17 @@ const int32_t kReserveNum = 1024 * 4;
 bool RebuildIndexTask::check() {
   return env_->kvstore_ != nullptr;
 }
+void RebuildIndexTask::finish(nebula::cpp2::ErrorCode rc) {
```
Comment (pengweisong) [inline src/storage/admin/RebuildIndexTask.cpp:20]: why just add check if it is IndexState::FAILED in line 76? If it is, then could skip it.

**Comment by pengweisong** [comment]
Talked offline,  LGTM now, but could be improved later: use taskId as guard instead of IndexState. IndexState is only as lock instead of other usage.

**Review by pengweisong — APPROVED** [review]

**Pull request merged**